Python 3.8b2 and TestDatetime64Timezone, too many warnings.

Running tests with python3.8b2+ on travis fails with two instead of one warning for TestDatetime64Timezone
```
E           AssertionError: 2 warnings found but 1 expected.
E           {message : DeprecationWarning('parsing timezone aware datetimes is deprecated; this will raise an error in the future'), category : 'DeprecationWarning', filename : '/home/travis/build/numpy/numpy/builds/venv/lib/python3.8/site-packages/numpy/core/tests/test_deprecations.py', lineno : 89, line : None}
E           {message : DeprecationWarning('an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.'), category : 'DeprecationWarning', filename : '/home/travis/build/numpy/numpy/builds/venv/lib/python3.8/site-packages/numpy/core/tests/test_deprecations.py', lineno : 89, line : None}
```
I think this problem is on the Python end in the `timedelta` class. I note that the function on our end looks buggy also, assigning a long to an int. The function where this occurs is deprecated, but until it is removed we should fix it.

EDIT: we might just want to mark it `xfail(python > 3.7)` if the problem really is on the Python end, in which case it should be fixed at some point. But let's wait until the first or second 3.8 rc before taking that step.
<!-- Please describe the issue in detail here, and fill in the fields below -->

### Reproducing code example:

<!-- A short code example that reproduces the problem/missing feature. It should be
self-contained, i.e., possible to run as-is via 'python myproblem.py' -->

```python
import numpy as np
<< your code here >>
```

<!-- Remove these sections for a feature request -->

### Error message:

<!-- If you are reporting a segfault please include a GDB traceback, which you
can generate by following
https://github.com/numpy/numpy/blob/master/doc/source/dev/development_environment.rst#debugging -->

<!-- Full error message, if any (starting from line Traceback: ...) -->

### Numpy/Python version information:

<!-- Output from 'import sys, numpy; print(numpy.__version__, sys.version)' -->

--- a/numpy/core/src/multiarray/datetime.c
+++ b/numpy/core/src/multiarray/datetime.c
@@ -2272,7 +2272,10 @@ convert_pydatetime_to_datetimestruct(PyObject *obj, npy_datetimestruct *out,
             if (tmp == NULL) {
                 return -1;
             }
-            seconds_offset = PyInt_AsLong(tmp);
+            /* Rounding here is no worse than the integer division below.
+             * Only whole minute offsets are supported by numpy anyway.
+             */
+            seconds_offset = (int)PyFloat_AsDouble(tmp);
             if (error_converting(seconds_offset)) {
                 Py_DECREF(tmp);
                 return -1;

